src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(MakeLocalFunctionStaticCodeRefactoringProvider)), Shared]
+    internal sealed class MakeLocalFunctionStaticCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, textSpan, cancellationToken) = context;
+
+            var localFunction = await context.TryGetRelevantNodeAsync<LocalFunctionStatementSyntax>().ConfigureAwait(false);
+            if (localFunction == null)
+            {
+                return;
+            }
+
+            if (!MakeLocalFunctionStaticHelper.IsStaticLocalFunctionSupported(localFunction.SyntaxTree))
+            {
+                return;
+            }
+
+            if (localFunction.Modifiers.Any(SyntaxKind.StaticKeyword))
+            {
+                return;
+            }
+
+            var semanticModel = (await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false))!;
```

Comment:
>(await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false))!; [](start = 32, length = 81)

This syntax looks so ugly and unnecessary, especially given this should never be null for C#/VB documents. This clearly seems to be a place where nullability is hurting readability of code. Can we define a separate extension method, say `GetNonNullSemanticModelAsync` or something that has this suppression operator?

Reply:
Agreed. I have created an issue for `GetDocument`, and this falls into the same category. I think we need a identify all usage of `!` in Roslyn and examine involved APIs to see if it makes sense to add a `GetNonNull` version of it.

Reply:
(this needs a companion in MS.CA.Analyzers if new API/extension method is added)